samples/apps/copilot-chat-app/webapi/Services/AzureContentModerator.cs

```diff
@@ -0,0 +1,145 @@
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI;
+using SemanticKernel.Service.CopilotChat.Options;
+
+namespace SemanticKernel.Service.Services;
+
+public record AnalysisResult(
+    [property: JsonPropertyName("category")] string Category,
+    [property: JsonPropertyName("riskLevel")] short RiskLevel
+);
+
+public record ImageContent([property: JsonPropertyName("content")] string Content);
+
+public record ImageAnalysisRequest(
+    [property: JsonPropertyName("image")] ImageContent Image,
+    [property: JsonPropertyName("categories")] List<string> Categories
+);
+
+public sealed class AzureContentModerator : IDisposable
+{
+    private const string HttpUserAgent = "Copilot Chat";
+
+    private readonly Uri _endpoint;
+    private readonly HttpClient _httpClient;
+    private readonly HttpClientHandler? _httpClientHandler;
+
+    /// <summary>
+    /// Options for the content moderator.
+    /// </summary>
+    private readonly ContentModerationOptions? _contentModerationOptions;
+
+    /// <summary>
+    /// Gets the options for the content moderator.
+    /// </summary>
+    public ContentModerationOptions? ContentModerationOptions => this._contentModerationOptions;
+
+    private static readonly List<string> s_categories = new()
+    {
+        "Hate",
+        "Sexual",
+        "SelfHarm",
+        "Violence" };
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureContentModerator"/> class.
+    /// </summary>
+    /// <param name="endpoint">Endpoint for service API call.</param>
+    /// <param name="apiKey">The API key.</param>
+    /// <param name="contentModerationOptions">Content moderator options from appsettings.</param>
+    /// <param name="httpClientHandler">Instance of <see cref="HttpClientHandler"/> to setup specific scenarios.</param>
+    public AzureContentModerator(Uri endpoint, string apiKey, ContentModerationOptions contentModerationOptions, HttpClientHandler httpClientHandler)
+    {
+        this._endpoint = endpoint;
+        this._contentModerationOptions = contentModerationOptions;
+        this._httpClient = new(httpClientHandler);
+
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+
+        // HACK
+        this._httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", apiKey);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureContentModerator"/> class.
+    /// </summary>
+    /// <param name="endpoint">Endpoint for service API call.</param>
+    /// <param name="apiKey">The API key.</param>
+    /// <param name="contentModerationOptions">Content moderator options from appsettings.</param>
+    public AzureContentModerator(Uri endpoint, string apiKey, ContentModerationOptions contentModerationOptions)
```

Comment:
can we just call the other constructor?
```suggestion
    public AzureContentModerator(Uri endpoint, string apiKey, ContentModerationOptions contentModerationOptions): this(endpoint, apiKey, contentModerationOptions, new() { CheckCertificateRevocationList = true }) { }
```

Reply:
No, we need the explicit constructor overload it can accept a HttpClientHandler as a parameter and the responsibility of disposing it is delegated to the caller:

Reply:
so it sounds like `_httpClientHandler` needs to be stateful so that later it can be disposed? would the following work?
```suggestion
    public AzureContentModerator(Uri endpoint, string apiKey, ContentModerationOptions contentModerationOptions): this(endpoint, apiKey, contentModerationOptions, this._httpClientHandler = new() { CheckCertificateRevocationList = true }) { }
```

Reply:
Looks like we can't :/ 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/e5d69fca-57e2-448d-ab43-ddbbc1f5eb95)

Reply:
looks like we can't :/ 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/7590dadc-c6d6-489a-88d5-edbc65f00b56)